### ISSUE TITLE
`missing_const_for_fn` not triggered on non-`const` fn with dependency on a `const fn`

### ISSUE THREAD

**Issue by Rudxain** [rust-lang/rust-clippy#9492]
### Summary

My `xorsum` crate has a userland implementation of `div_ceil` and `next_multiple_of` (named `next_multiple` in my `module.rs`). Clippy correctly detected that `div_ceil` should be `const`, but it didn't show a warning for `next_multiple` (even after declaring `div_ceil` as `const`).

Maybe it's `match`'s fault?

### Lint Name

missing_const_for_fn

### Reproducer

I tried this code:

```rust
#![deny(clippy::missing_const_for_fn)]

#[inline]
const fn div_ceil(n: usize, d: usize) -> usize {
	match (n / d, n % d) {
		(q, 0) => q,
		(q, _) => q + 1,
	}
}

#[inline]
fn next_multiple(n: usize, d: usize) -> usize {
	match d {
		0 => d,
		_ => div_ceil(n, d) * d,
	}
}
```

I expected to see this happen: an error about `next_multiple`.

Instead, this happened: nothing

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-unknown-linux-gnu
release: 1.63.0
LLVM version: 14.0.5
```

**Comment by lukaslueg** [comment]
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f62d12f1d2f9af249ad01657752117b7) lints on `next_multiple` with the example above.

It won't trigger on `next_multiple` unless `div_ceil` has already been made `const` first, as it does not take into account that `div_ceil` _could_ be `const` when analyzing `next_multiple`.

**Comment by Rudxain** [comment]
> It won't trigger on `next_multiple` unless `div_ceil` has already been made `const` first, as it does not take into account that `div_ceil` _could_ be `const` when analyzing `next_multiple`.

I know, I've read the docs (thanks for the info anyways). The actual problem is that it doesn't trigger (on `next_multiple`) **even after** declaring `div_ceil` as `const`

**Comment by lukaslueg** [comment]
Excuse me if I still misunderstand: Clippy's version on the playground (see link above) correctly triggers on `next_multiple`... ?!

**Comment by Rudxain** [comment]
> Excuse me if I still misunderstand: Clippy's version on the playground (see link above) correctly triggers on `next_multiple`... ?!

Oh right, I'm sorry, I forgot to open Tools then Clippy. That's true, it does trigger in the playground, but not on my machine when I have [this file](https://github.com/Rudxain/xorsum/blob/main/src/module.rs) opened\*

\* The link isn't a permalink, it points to the most recent commit, which already has both fns declared as const. I can't remember which commit corresponds to the playground, but it's also recent. I'll try to find it now

**Comment by lukaslueg** [comment]
Correct me if I'm wrong, but the lint seems to work as expected. Maybe you can provide another reproducible example?

**Comment by Rudxain** [comment]
I'm going to attempt to recreate the bug in another repo branch. I'll share the link when it's ready.

BTW, I forgot to mention that I'm using `cargo clippy` via VScode, and I'm also using the `rust-analyzer` extension, so maybe one of them is causing some conflict? I also didn't run `cargo clean`, which means that the cache may prevent clippy from noticing changes (I guess? Please correct me if I'm wrong)

**Comment by Rudxain** [comment]
This is the branch: https://github.com/Rudxain/xorsum/tree/Clippy-MCFF-test

But the bug is fixed now, I guess I had to run `cargo clean`. I'm sorry for any inconvenience. I'll close the issue.

> Note: Since the branch seems useless now, I'll delete it in less than a week (unless anybody requests otherwise)

**Comment by lukaslueg** [comment]
No worries, feel free to reopen if this is reproducible
